index.d.ts

```diff
@@ -69,10 +68,8 @@ export type CamelCaseKeys<
 				true,
 			]
 				? T[P]
-				// eslint-disable-next-line @typescript-eslint/ban-types
-				: {} extends CamelCaseKeys<T[P]>
```

Comment:
This has been added in [this PR](https://github.com/sindresorhus/camelcase-keys/pull/92), which is essentially a `Record<string, any>`, which would have been `{}` due to the fact that it is a `Record<string, any>`. This change to `unknow` makes this condition unnecessary.

```ts
// type result is 'true'
type test<T> = T extends Record<string, any> ? 'true' : 'false';
type result = test<() => 123>;
```